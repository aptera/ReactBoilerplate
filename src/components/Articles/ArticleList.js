import React, {Component} from 'react';
import Article from './Article';
import values from 'object.values';
import DataApi from './services/DataApi';
import { data } from './testData';

class ArticleList extends Component {

    constructor(props) {
        super(props);
        const api = new DataApi(data);
        this.state = {
            articles: api.getArticles(),
            authors: api.getAuthors()
        };
    }
    
    articleActions = {
        lookupAuthor: (authorId) => this.state.authors[authorId],
    }

    render(){
        return (
            <div>
                {values(this.state.articles).map((article) => 
                    <Article
                        key={article.id}
                        article={article}
                        actions={this.articleActions}
                    />
                )}
            </div>
        );
    } 
}

export default ArticleList;