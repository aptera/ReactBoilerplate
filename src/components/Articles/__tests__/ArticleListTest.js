import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {

    const testProps = {
        data: {
            a: { id: 'a' },
            b: { id: 'b' },
            c: { id: 'c' },
        }
    };

    it('renders correctly', () => {

        const tree = renderer.create(
            <ArticleList
                data={testProps.data}
            />
        ).toJSON();
        expect(tree.children.length).toBe(3);
        expect(tree).toMatchSnapshot();
    });
});
