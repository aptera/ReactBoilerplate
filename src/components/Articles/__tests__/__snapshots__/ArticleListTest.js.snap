// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    className="article"
  >
    <h2>
      Article One
    </h2>
    <div
      className="date"
    >
      01/01/2018
    </div>
    <div>
      <a
        className="website"
        href="https://github.com/dfederspiel"
      >
        David
         
        Federspiel
      </a>
    </div>
    <div
      className="body"
    >
      This is an article about Ones's
    </div>
  </div>
  <div
    className="article"
  >
    <h2>
      Article Two
    </h2>
    <div
      className="date"
    >
      01/05/2018
    </div>
    <div>
      <a
        className="website"
        href="https://github.com/dfederspiel"
      >
        David
         
        Federspiel
      </a>
    </div>
    <div
      className="body"
    >
      This is an article about Two's
    </div>
  </div>
  <div
    className="article"
  >
    <h2>
      Article Three
    </h2>
    <div
      className="date"
    >
      02/06/2018
    </div>
    <div>
      <a
        className="website"
        href="https://github.com/dfederspiel"
      >
        David
         
        Federspiel
      </a>
    </div>
    <div
      className="body"
    >
      This is an article about Three's
    </div>
  </div>
</div>
`;
